README.md

```diff
@@ -219,6 +219,37 @@ Default setup uses CPU (`--ngpu 0`).
 Note that if you want to use multi-gpu, the installation of [nccl](https://developer.nvidia.com/nccl)
 is required before setup.
 
+
+### Changing the configuration
+The default configurations for training and decoding are written in `conf/train.yaml` and `conf/decode.yaml` respectively.  It can be overwritten by specific arguments: e.g.
+
+```bash
+asr_train.py --config conf/train.yaml --batch-size 24
+```
+
+In this way, you need to edit `run.sh` and it might be inconvenient.
+Instead of giving arguments directly, we recommend you to modify the yaml file and give it to `run.sh`:
+
+```bash
+./run.sh --train-config conf/train_modified.yaml
+```
+
+We also provide a utility to modify a yaml file from command line:
+
+```bash
+<conf/train.yaml change_yaml.py -a batch-size=24 > conf/train_batchsize24.yaml && ./run.sh --train-config conf/train_batchsize24.yaml
```

Comment:
Maybe this is too much for your PR but I wanna request auto naming to make CLI shorter. For example
```
# this line saves new file named "conf/train_batch-size24.yaml"
# and print its new name to stdout
$ change_yaml.py conf/train.yaml -a batch-size=24 
conf/train_batch-size24.yaml

# therefore we can simply do
$ ./run.sh --train-config $(change_yaml.py conf/train.yaml -a batch-size=24)
```

Reply:
If there were auto naming, we do not repeat those names in hyper parameter search
```bash
# before
for lr in 0.1 0.01 0.001; do
  for bs in 8 16 32; do
    <conf/train.yaml change_yaml.py -a batch-size=$bs -a lr=$lr > conf/train_batchsize$bs_lr$lr.yaml
    ./run.sh --train-config conf/train_batchsize$bs_lr$lr.yaml
  done  
done
```
```bash
# after
for lr in 0.1 0.01 0.001; do
  for bs in 8 16 32; do
    ./run.sh --train-config $(change_yaml.py conf/train.yaml -a batch-size=$bs -a lr=$lr)
  done  
done
```

Reply:
I like @ShigekiKarita ’s idea.
But how do we define the name of `expdir`?
(Currently `basename $train_config .*`)

Reply:
https://github.com/espnet/espnet/blob/d1e75dbad11b22c9f1f2debb6fc623cbc9719116/egs/aishell/asr1/run.sh#L187-L196

Do you mean these lines? If `new_yaml.py` emits `conf/train_xxx.config` and print its name to the `--train_config $(new_yaml.py ...)`, then `run.sh` creates the expdir at `exp/${train_set}_${backend}_train_xxx/` by default. I do not think this is problem in my usage. Would you describe more what you concern about?

Reply:
Sorry, I misunderstood that `$(new_yaml.py ...)` is standard output of new yaml file.
It is no problem.

Reply:
@ShigekiKarita 's looks fine. I adopted it in change_yaml.py.